### PR TITLE
avoid extra sha1_hash memory copy, minor formatting

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/create_torrent.hpp [hunk 1]
```diff
@@ -366,8 +366,7 @@ namespace libtorrent {
 		bool m_include_symlinks:1;
 	};
 
-namespace detail {
-
+	namespace detail {
```
Comment (arvidn) [inline include/libtorrent/create_torrent.hpp]: I'm actually hoping to go the other direction, where namespaces don't add additional indentation. (in some way to adopt the of C++17 feature of being able to open nested namespaces in a single namespace definition)

File: src/create_torrent.cpp [hunk 2]
```diff
@@ -179,7 +179,7 @@ namespace {
 				st->iothread.abort(true);
 				return;
 			}
-			st->ct.set_hash(piece, sha1_hash(piece_hash));
```
Comment (arvidn) [inline src/create_torrent.cpp]: this was just a copy constructor, it didn't actually hash the digest, right?

File: include/libtorrent/create_torrent.hpp (hunk 1 above)
Comment (aldenml) [inline include/libtorrent/create_torrent.hpp]: ok, I will move the `}` to the start line then :)

File: src/create_torrent.cpp (hunk 2 above)
Comment (aldenml) [inline src/create_torrent.cpp]: yes, just a copy constructor

File: include/libtorrent/create_torrent.hpp (hunk 1 above)
Comment (aldenml) [inline include/libtorrent/create_torrent.hpp]: actually, `inline void nop(piece_index_t) {}` is already double indented, do you really prefer remove one indentation level here?

Comment (arvidn) [inline include/libtorrent/create_torrent.hpp]: well, I would prefer no more than one indentation level as the end goal (and probably just make namespaces not indent at all). Small corrections in that direction are fine, I'm procrastinating a bit about changing all the indentations in all files. I don't think it's urgent and it comes with some (merge) risk, so I don't know about that the best approach is there.

in this case, if it's just the ``nop`` function, please fix its indentation

Comment (aldenml) [inline include/libtorrent/create_torrent.hpp]: done

**Pull request merged**